docs/askpass-url.md

```diff
@@ -0,0 +1,40 @@
+# Using an Http Auth URL with git-sync
+
+## Step 1: Create a GIT_ASKPASS HTTP Service
+
+The GIT ASKPASS Service expose via HTTP and provide the answer to GIT_ASKPASS.
+
+Example of the servcie's output, see more at <https://git-scm.com/docs/gitcredentials>
+
+```json
+username=xxx@example.com
+password=ya29.mysecret
+```
+
+## Step 2: Configure git-sync container
+
+In your git-sync container configuration, specify the GIT_ASKPASS_URL
+
+The credentials will pass in plain text, make sure the connection between git-sync
+and GIT ASKPASS Service are secure.
+
+The recommended way is the ASKPASS Service running within the same pod as git-sync.
+
+See <https://github.com/cydu-cloud/git-askpass-gce-node> as a full example which use GCE Node Service Account credential to access Google Cloud Source Repo.
```

Comment:
We should find a better place to put this.. maybe GoogleCloudPlatform's repo?

Reply:
Can we cook down an e2e test that can be set up and run locally?  E.g. spin up a git server and an HTTP server, tell git-sync to pull the password from the HTTP server.

test_e2e.sh should cover this

Reply:
For the e2e test, it depends on askpass_git test I just created in another PR, don't want the PR grow into too big, can I add it as follow up PR? This won't hurt any existing functionality even the callbalk is broken.

For GoogleCloudPlatform's repo, it might take a while to setup, need to go through some process.

Reply:
I don't want to e2e against GCP.  I want to set up a local git repo that demands user/pass and a local HTTP server which serves the content.  I am working on a more complex SSH example - will link it here, hopefully helps you with infra

Reply:
For "local git repo that demands user/pass", trying to use askpass_git as I did for user/pass e2e test, the `git credential fill` is exactly what git do when demands user/pass, so the simulation should be very good enough for the test here.  
Removed the reference docs to gcp. It's purely local e2e test right now.

Thanks for the SSH example, I can not image any better way to do it for SSH, but for user/pass, I personally prefer askpass_git a bit more since it's light and easier to debug/maintain. 

Make sense? 

Sorry for the growing PR, I just manually merged the askpass_git from https://github.com/kubernetes/git-sync/pull/217.